routers/api/v1/repo/release.go

```diff
@@ -62,42 +61,30 @@ func CreateRelease(ctx *context.APIContext, form api.CreateReleaseOption) {
 		ctx.Status(404)
 		return
 	}
-	tag, err := ctx.Repo.GitRepo.GetTag(form.TagName)
+	rel, err := models.GetRelease(ctx.Repo.Repository.ID, form.TagName)
 	if err != nil {
-		ctx.Error(500, "GetTag", err)
+		if models.IsErrReleaseNotExist(err) {
```

Comment:
I'm confused here. If we are creating a release, why do we expect there to already be a release with the name `form.TagName`?

Reply:
because release table does now contains also tags that are not full releases with attachments etc and when creating release it is needed to update release table flag (is_tag to false) and adding additional release information to that row